app/models/course.rb

```diff
@@ -28,6 +28,7 @@ class Course < ApplicationRecord
 
   has_associated_audits
   audited except: :changed_at
+  validates :course_code, uniqueness: { scope: :provider_id }
```

Comment:
Looking at `schema.rb` I don't think `course_code` is indexed. This is going to be a bit of a performance hit, I think we should add it.

We might also consider adding a unique index on the combination of `course_code` and `provider_id`.

Reply:
My mistake, the index is there I just missed it.